### DISCUSSION TITLE
Watch mode with HMR without reporter-dev-server

### DISCUSSION THREAD

**Discussion by zouloux** [parcel-bundler/parcel#6088] [General]
Hi all, thanks a lot for all the work done on Parcel 2 :)
I'm using Parcel 2 mainly with the API (not the CLI), and I noticed something weird. Maybe it is a feature and not a bug so this is why i'm not opening an issue.

If I got this config : 

```
const bundler = new Parcel({
	entries: 'src/test/index.html',
	targets: {
		app: {
			distDir: 'dist/',
		}
	},
	hmrOptions: {
		port: 1234,
		host: 'localhost'
	},
	mode: 'development',
	additionalReporters: [
		{ packageName: '@parcel/reporter-cli', resolveFrom: __filename },
	],
});
```

HMR runtime is injected into generated bundle. But, HMR port is not opened, it took me some long minutes to catch that the HMR does not connect because the HMR service is actually not running.
Here is the error in chrome dev tools.
```
WebSocket connection to 'ws://localhost:1234/' failed: 
[parcel] ðŸš¨ Connection to the HMR server was lost
```

I inspected what config parcel cli was using (with good old console log) and discovered that adding `{ packageName: '@parcel/reporter-dev-server', resolveFrom: __filename }` resolved the issue, HMR is correctly connecting.

To me 'watch' should start HMR if we have an `hmrOptions`, other options should not change this behavior, this is a bit misleading.

I know you are working pretty hard on the documentation also and maybe I missed something.
( working with Parcel version beta-2, tested others versions and it behaved like so )

I can create a ticket if needed :)
Thanks a lot,
Alexis

**Comment by mischnic** [comment]
This is indeed misleading. Do you want to open a PR that removes
https://github.com/parcel-bundler/parcel/blob/ff1a55e8a8d0ec989adfede698771b7fdb567d95/packages/core/parcel/src/cli.js#L452

and adds that plugin with `reporters: ["@parcel/reporter-dev-server"]` in https://github.com/parcel-bundler/parcel/blob/v2/packages/configs/default/index.json?

**Reply by zouloux** [reply]
Cool ! Indeed it would be better to have it embedded in default config.

I understand why the cli-reporter is not included into default config but rather injected at cli.js runtime. But I do not get why HMR server does not start without reporter-dev-server. I'm sure tho you have a good reason to make it like so :)

Anyway, it would also be great to update the doc to explain this specific case, or at least throw something in Parcel if users tries to configure HMR without reporter-dev-server.

Will do the PR soon, keeping you updated.

Thanks a lot for your quick reply :)

**Reply by mischnic** [reply]
> But I do not get why HMR server does not start without reporter-dev-server. I'm sure tho you have a good reason to make it like so :)

I removed it from the default config myself and I forgot about the JS API usecase 😄 

>  at least throw something in Parcel if users tries to configure HMR without reporter-dev-server.

That's not really possible with the plugin architecture (for example you might want to use your own dev server plugin that uses these options).

**Reply by mischnic** [reply]
-> https://github.com/parcel-bundler/parcel/pull/6195

**Reply by zouloux** [reply]
Thanks, sorry didn't had time to do it !

**Reply by mischnic** [reply]
No worries!

**Reply by zouloux** [reply]
Cool, but I missed to contribute to this amazing project :) thanks again
